src/page/MainContent.jsx

```diff
@@ -69,24 +75,56 @@ const MainH1 = styled.h1`
 `;
 
 const MainContent = ({
-  bestProducts,
   sortedProducts,
   sortOrder,
   setSortOrder,
   activePage,
   setActivePage,
-  totalpage,
+  totalPage,
 }) => {
+  useEffect(() => {
+    const mediaQuery = window.matchMedia("(max-width: 768px)");
+    const fetchBestProducts = async () => {
+      const pageSize = mediaQuery.matches ? 2 : 4;
+      try {
+        const data = await fetchProducts(1, "favorite", pageSize);
+        setBestProducts(data.list);
+      } catch (error) {
+        console.error("Error fetching best products:", error);
+      }
+    };
+    fetchBestProducts();
+  }, []);
+
+  const [bestProducts, setBestProducts] = useState([]);
+  const [filterProducts, setFilterProducts] = useState([]);
+
+  useEffect(() => {
+    setFilterProducts(sortedProducts);
+  }, [sortedProducts]);
+
+  const handleSearchChange = (e) => {
+    const inputValue = e.target.value;
+
+    const filterProduct = sortedProducts.filter((element) => {
+      return element.name.startsWith(inputValue);
+    });
+    setFilterProducts(filterProduct);
+  };
```

Comment:
P3:
변수나 state는 최상단으로 배치해주시는게 가독성에 더 좋습니다.

```suggestion
  const [bestProducts, setBestProducts] = useState([]);
  const [filterProducts, setFilterProducts] = useState([]);

  const handleSearchChange = (e) => {
    const inputValue = e.target.value;

    const filterProduct = sortedProducts.filter((element) => {
      return element.name.startsWith(inputValue);
    });
    setFilterProducts(filterProduct);
  };
  
  useEffect(() => {
    const mediaQuery = window.matchMedia("(max-width: 768px)");
    const fetchBestProducts = async () => {
      const pageSize = mediaQuery.matches ? 2 : 4;
      try {
        const data = await fetchProducts(1, "favorite", pageSize);
        setBestProducts(data.list);
      } catch (error) {
        console.error("Error fetching best products:", error);
      }
    };
    fetchBestProducts();
  }, []);

  useEffect(() => {
    setFilterProducts(sortedProducts);
  }, [sortedProducts]);
```